T4/IncludeHeader.ttinclude

```diff
@@ -288,5 +294,81 @@ namespace <#=Namespace#>.Include
         return new UsingNamespace { Namespace = ns ?? "", };
     }
 
+    static readonly Regex s_typeClassificationWithinCommentOrQuotes = new Regex(
+        "\".*?(struct|class|enum).*?\"|//.+?(struct|class|enum).*?",
+        RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.Singleline
+        );
+
+    static string[] Internalize(string[] lines)
+    {
+        var depth = 0;
+        var namespaceDepth = 0;
+        var processed = new string[lines.Length];
+        var lineIndex = 0;
+
+        foreach (var line in lines)
+        {
+            var wasProcessed = false;
 
+            try
+            {
+                if (line.Contains("namespace"))
```

Comment:
Yes that would be a problem actually. Could probably use a regex similar to `s_typeClassificationWithinCommentOrQuotes` to make sure it's not within quotes or in a comment.